### PR TITLE
feat: Progress bar length and font size increases for webui

### PR DESCRIPTION
**Pull request by GaryElshaw**
Increased font legibility with `small` as the baseline. The progress bar is increased to 100px in compact view.

A successor to #5030. Addresses some of the issues raised in #4994. 

The current compact UI:
Screenshot from 2024-01-02 15-42-10

This pr:

Screenshot from 2024-01-02 15-22-40

Notes: Increased base font sizes, and progress bar size in compact view

### CODE REVIEW

**Comment by GaryElshaw** [comment]
Are there objections?

**Comment by Rukario** [comment]
It looks like the current UI has already accomplished similar width as your 2nd screenshot now, there must be some other PR doing that.

Looking at the code, there are 10vw for width meaning 10% of viewport in width, let's do some scenario with a 1080p monitor; 10vw of browser maximized on 1080p monitor would've given you 192 pixels.

50px for min-width meaning width must be at least 50px, when the width goes under this threshold the 50px will come into place.

If you give it 250px the width 10vh will become next to useless because 250 is greater than 192, only usable on 1440p and greater. And narrowed browsers will be having hard time with 250px progress bars eating up too much space into torrent details.

**Comment by GaryElshaw** [comment]
Here's 190px and the increased font size.

image

**Comment by Rukario** [comment]
Now you're emulating the width of a 1080p monitor 😝

I'm just saying the number 192 is 10% of 1080p (1920x1080). You should also realize what does the `min-width` do and once you hardcoded this number into it, it's not going to let smaller browser shrink it any further.

**Comment by GaryElshaw** [comment]
I think that's fine, but i'm open to suggestions. 😝 yourself! :-)

**Comment by Rukario** [comment]
I recommend 100px progressbar, it's a perfect number to use for percentage based progress bar, advancing pixels will be 1:1 with percent digits!

I'd prefer the font to be consistent across full and compact mode but I do like how you made labels bigger for compact mode, it became readable, I'll want to have that for full mode. Although it's became readable but maybe too big, 0.75em should do it.

I'm sure there will be another way to make up for the big empty space during compact mode, as @Pentaphon suggested, panel mode sounds like a great way to make up for that. It's a behavior of old Web UI not making it over to the overhauled web UI.

**Comment by GaryElshaw** [comment]
Knocked the progress bar back to 100, but i'm definitely preferring 0.8 over 0.7.

0.7

Screenshot from 2023-08-22 16-43-44

0.8

Screenshot from 2023-08-22 16-44-51

**Comment by Rukario** [comment]
Oh, no no, I mean 0.75em for labels where the `x-small` has been, it's a middle ground between `small` and `x-small`, also if you said 0.8 without secondary fractional digit it is what `small` would've been.

At the very least the progress bar is 100px meaning small browsers can at least have remaining space for other torrent details. Would you get rid of that `width: 10vw` also so us on large monitors not jealous of small screen's perfect 100px bar?

**Comment by GaryElshaw** [comment]
> if you said 0.8 without secondary fractional digit it is what `small` would've been.

Yep, sorry, i was meaning 0.875.

Made the changes from x-small to 0.75em and removed `width: 10vw`

**Comment by GaryElshaw** [comment]
@rukario @dareiff Good to go?

**Comment by Pentaphon** [comment]
Thanks for following up on this, Gary. We really needed a much more legible compact mode for desktop browsers, which are probably the most used form of WebUI from what I can see. It was just way too tiny. There was clearly too much focus on having it work on mobile rather than desktop. 

> I'm sure there will be another way to make up for the big empty space during compact mode, as @Pentaphon suggested, panel mode sounds like a great way to make up for that. It's a behavior of old Web UI not making it over to the overhauled web UI.

Thanks for supporting this! I really think a paneled UI is the only way forward for desktop browser WebUI. It's just never going to look right on the desktop, even with the effort Gary has put in to make it more readable.

**Review by dareiff — APPROVED** [review]
LGTM!

**Comment by Rukario** [comment]
This PR
Transmission ga 1

Original
Transmission ga 2

Animated
Transmission ga 1 (apng)

Likes:
- 100px progress bar
- 0.75em label seen in full mode

Dislikes:
- Everything else 🫠

**Comment by Pentaphon** [comment]
> I know I don't usually give UI feedback, but I'm not a fan of making the torrent names bold in compact mode like this.
> 
> It makes more sense in the default mode since it distinguishes from other nearby text (e.g. status and transfer speed); but in compact mode, a list of torrents becomes just a wall of boldface e.g. in the right-hand-side of [this screenshot](https://user-images.githubusercontent.com/2275021/262770254-69e53a62-3e5a-4de8-849c-349d74b13702.png).

Yeah, you probably have a point there since the screenshot only shows 3 lines instead of 100 torrents. 

@GaryElshaw as long as the font sizes are tweaked to make the compact torrent list more readable, I think you can do without the bold torrent titles in compact mode. After all, I think we all imagine compact mode to look more like a traditional torrent list as you see in uTorrent, qBittorrent, etc.

**Comment by GaryElshaw** [comment]
Betterer?

Screenshot from 2023-09-06 09-13-55-obfuscated

**Comment by Pentaphon** [comment]
> Betterer?

Yep. Thanks! This definitely fixes all the issues compact mode had.

File: web/assets/css/transmission-app.scss [hunk 1]
```diff
@@ -523,8 +523,8 @@ $video-image: '../img/film.svg';
     }
 
     .torrent-labels {
-      font-size: x-small;
-      font-weight: normal;
+      font-size: 0.75em;
+      font-weight: bold;
```
Comment (Rukario) [inline web/assets/css/transmission-app.scss:527]: Please make font weight normal again for labels in full mode

File: web/assets/css/transmission-app.scss [hunk 2]
```diff
@@ -534,7 +534,8 @@ $video-image: '../img/film.svg';
 
       &.compact {
         flex: 1;
-        font-size: x-small;
+        font-size: 0.875em;
+        font-weight: normal;
```
Comment (Rukario) [inline web/assets/css/transmission-app.scss:538]: Erase all in this selector leaving only flex: 1;

**Comment by GaryElshaw** [comment]
This is good to go.

File: web/assets/css/transmission-app.scss [hunk 3]
```diff
@@ -553,7 +554,8 @@ $video-image: '../img/film.svg';
     .torrent-progress-details,
     .torrent-peer-details {
       color: var(--color-fg-primary);
-      font-size: small;
+      font-size: 0.875em;
+      font-weight: normal;
```
Comment (Rukario) [inline web/assets/css/transmission-app.scss:558]: This font-weight has no effect so should be safe to remove

I also want to make the font small again for progress and peer details.

**Review by Rukario — CHANGES_REQUESTED** [review]
Frell! I forgot to submit review

**Comment by Rukario** [comment]
Please take your time and apply change suggestions, although there are a lot of font sizing issues resolved already for #4994, this PR is probably the last PR to cross out all the font sizing issues. I like how this PR is sunsetting any use of "x-small" for font-size. Labels are becoming readable which was one of the "x-small".

File: web/assets/css/transmission-app.scss (hunk 1 above)
Comment (GaryElshaw) [reply web/assets/css/transmission-app.scss:527]: Done

File: web/assets/css/transmission-app.scss (hunk 2 above)
Comment (GaryElshaw) [reply web/assets/css/transmission-app.scss:538]: Okay

File: web/assets/css/transmission-app.scss (hunk 3 above)
Comment (GaryElshaw) [reply web/assets/css/transmission-app.scss]: I've made that change, but i think normal should be the baseline. I don't see any reason for using small or x-small in this interface - at all.

Comment (Rukario) [reply web/assets/css/transmission-app.scss]: If you're fine with "small" and you don't see the issue with going bigger yet- doesn't mean it's not going to be an issue.

**Comment by ckerr** [comment]
@Rukario looks like there have been commits to address your feedback; please re-review :bow:

**Comment by GaryElshaw** [comment]
@Rukario ping

**Comment by Rukario** [comment]
I'd like some more discussion/opinion from other people regarding font size used for peer and progress detail https://github.com/transmission/transmission/pull/5340#discussion_r1322475416

I'm not sold on the font size used for peer/progress detail, may I suggest a martial merge at the meantime? Thank you!

**Comment by GaryElshaw** [comment]
Any thoughts as to why this might be a conflict?

Screenshot from 2024-01-01 15-23-12

**Comment by ckerr** [comment]
@Rukario looks like the current state of the PR removes all the changes that [you had questions about](https://github.com/transmission/transmission/pull/5340#issuecomment-1815268375) so I _think_ this is ok to merge?

My own opinion, FWIW:
- I'm much happier with a 100px max progressbar than the original 250px
- I don't have any personal opinion about x-small vs small, but if this finishes sunsetting x-small then I'm fine with it.

**Pull request merged**